plugin/src/types.ts

```diff
@@ -10,6 +10,7 @@ export interface ReanimatedPluginOptions {
   relativeSourceLocation?: boolean;
   disableInlineStylesWarning?: boolean;
   processNestedWorklets?: boolean;
+  omitWorkletInitData?: boolean;
```

Comment:
I understand it will be mostly used in the preset, but still I think we should allow for a more fine-grained control for the user. `omitWorkletInitData` seems mysterious, a user might ask himself: 'what is it and should I really omit it?'. I think we should really signify with the name of that option, that enabling it omits data that is required only on native platforms.

Reply:
Do you mean `omitWorkletInitData_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`? 🤔

Reply:
> Do you mean `omitWorkletInitData_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`? 🤔

what

Reply:
reminded me of
![image](https://github.com/software-mansion/react-native-reanimated/assets/40713406/8cabc869-4b70-4bd3-8dd9-723c75e258d6)